src/mscorlib/shared/System/IO/Path.cs

```diff
@@ -477,6 +477,22 @@ public static bool TryJoin(ReadOnlySpan<char> path1, ReadOnlySpan<char> path2, R
             return true;
         }
 
+        public static bool TryGetTempPath(Span<char> destination, out int charsWritten)
+        {                        
+            charsWritten = 0;
+
+            var tmpPath = Path.GetTempPath();
```

Comment:
i want to avoid this allocation if `destination < tmpPath.Length`, on Windows i could use `ValueStringBuilder` but on unix side we don't have 
```cs
private static void GetTempPath(ref ValueStringBuilder builder)
```
on `Path.Unix.cs`  there is 
```cs
public static string GetTempPath()
```
with `Environment.GetEnvironmentVariable(TempEnvVar)` string allocation.
is there `private static string GetEnvironmentVariableCoreHelper(string variable, Span<char> buffer)` for [unix](https://source.dot.net/#System.Private.CoreLib/src/System/Environment.cs,451)?

Reply:
> is there private static string GetEnvironmentVariableCoreHelper(string variable, Span<char> buffer) for unix?

As part of having this we should write the lower level pieces we need. :)  I'd make an internal one that takes a ValueStringBuilder.  I'm looking into what it would take to polish up that class and make it public in the future, btw.

Reply:
@JeremyKuhne i get the point, can you help me to find where `Environment.GetEnvironmentVariable()` pinvoke for unix are defined?If i follow code with VS i reach Win32Native only.

Reply:
In this case the actual P/Invoke is provided by the PAL. So the same code is used in both places.  The cross-plat model the native parts of the runtime uses is that it has Unix implementations for all of the Win32 APIs. It simply includes the PAL when building for Unix instead of the Windows headers/libs. In the managed code this abstraction is also leveraged. Anything in `Win32Native` has a Unix implementation. The source file that is included when building for Unix is `coreclr\src\mscorlib\src\Interop\Unix\Interop.Libraries.cs`.

As such, you can use this straight-up: https://github.com/dotnet/coreclr/blob/57ee22e6086af5dfc06651dc002e24a71c3999ee/src/mscorlib/src/Microsoft/Win32/Win32Native.cs#L365

Reply:
TYVM